readme.md

```diff
@@ -54,6 +54,47 @@ Type: `Function`
 
 The callback function to execute when the process exits.
 
+### exitHook.async(asyncHookOptions)
+
+Returns a function that removes the hook when called. Please see [Async Notes](#async-notes) for considerations when using the asynchronous API.
```

Comment:
This should have a tl;dr (summary). The async notes are quite verbose and few will bother to read it all. Like, it's not guaranteed to be called, use `gracefulExit()`, etc.